src/coreclr/System.Private.CoreLib/src/System/Enum.CoreCLR.cs

```diff
@@ -11,7 +11,6 @@ namespace System
 {
     public abstract partial class Enum
     {
-        // This returns 0 for all values for float/double/nint/nuint.
```

Comment:
We can as well make this work while we are on it. It makes the code simpler and fixes #29266.

Reply:
(I have pushed commit to do that.)